crates/client-api/src/routes/database.rs

```diff
@@ -400,24 +439,23 @@ where
 {
     // Anyone is authorized to execute SQL queries. The SQL engine will determine
     // which queries this identity is allowed to execute against the database.
-
-    let db_identity = name_or_identity.resolve(&worker_ctx).await?;
-    let database = worker_ctx_find_database(&worker_ctx, &db_identity)
-        .await?
-        .ok_or(NO_SUCH_DATABASE)?;
-
-    let auth = AuthCtx::new(database.owner_identity, auth.identity);
+    let Connected {
+        caller_identity,
+        database,
+        leader,
+        module,
+        connection_id,
+    } = call_on_connect(worker_ctx, auth, name_or_identity).await?;
+
+    let auth = AuthCtx::new(database.owner_identity, caller_identity);
     log::debug!("auth: {auth:?}");
 
-    let host = worker_ctx
-        .leader(database.id)
-        .await
-        .map_err(log_and_500)?
-        .ok_or(StatusCode::NOT_FOUND)?;
-    let json = host.exec_sql(auth, database, body).await?;
+    let json = leader.exec_sql(auth, database, body).await?;
 
     let total_duration = json.iter().fold(0, |acc, x| acc + x.total_duration_micros);
 
+    call_on_disconnect(module, connection_id, caller_identity).await?;
+
```

Comment:
You shouldn't `?` on the result from `exec_sql` until after you call `call_on_disconnect`; otherwise, if the query fails for any reason, the module will never know that the client has disconnected. probably want something like

```rust
let json_result = exec_sql.await;
let disconn_result = call_on_disconnect.await;
let json = json_result?;
disconn_result?;
```

But that precise order of a sql error taking precedence over an on_disconnect error probably doesn't matter too much